### ISSUE TITLE
[Accessibility]: Tables in code excerpts should use headers for line numbers

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#43144]
### Audit type

ARC Toolkit

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34190

### Problem description

ArcToolkit says the tables used for search results are confusing because a11y tools can't tell if the table is tabular data or used for layout purposes. We need to clarify this.

### Expected behavior

The table should switch the line numbers from being `<td>` to `<th>`. This will clarify that this is indeed a table with tabular data. ArcToolkit should not show alerts about tables.

![CleanShot 2022-10-18 at 15 52 06@2x](https://user-images.githubusercontent.com/206864/196559884-ffcfcfa3-abd9-4aea-aa39-9d0ab67f766e.png)

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by GitStart. I have assigned the label 'gitstart' and added the assignee 'gitstart-sourcegraph'.

/cc @lguychard

**Comment by felixfbecker** [comment]
Reopening since https://github.com/sourcegraph/sourcegraph/pull/44514 was reverted

**Comment by limitedmage** [comment]
I'll work to restore the original PR and fix the issues that led to the revert.
